Run `cargo install --debug --locked` on the CI

qryxip opened the pull request:
~~Fixes # 54.~~

qryxip commented:
By the way, cargo-udeps v0.1.8 is not uploaded to crates.io.

est31 commented:
I'd rather wait for https://github.com/rust-lang-nursery/failure/pull/343 to be merged instead of fixing the quote version. The CI changes look ok though. Can you remove the Cargo.toml changes?

est31 commented:
> By the way, cargo-udeps v0.1.8 is not uploaded to crates.io.

Oh no. Thanks for the pointer, I've published it now manually.

est31 commented:
Also @qryxip the syn dependency can be removed but it still needs a cargo update to get the structopt update in. Could you make one?

est31 commented:
For reference, this was the structopt fix: https://github.com/TeXitoi/structopt/commit/bc43cdfb9252bcfb02e00fd0917ce1aa9dfc0480#diff-f8859fa7599bdffe4bea39369662f6f0

qryxip commented:
> it still needs a cargo update to get the structopt update in

That's true. I thought the version of `structopt` was still v0.3.7 or earlier.

|                            | `master` = `v0.1.8` | This PR   |
| -------------------------- | ------------------- | --------- |
| `locked`, `dev-deps`       | ✓                  | ✓        |
| `locked`, `avoid-dev-deps` | ✓                  | ~~X~~ ✓  |
| `update`, `avoid-dev-deps` | X                   | ~~✓~~ X  |

qryxip commented:
I removed the two commit that modify `Cargo.toml` and `Cargo.lock`. But `cargo install --locked` emits a warning (not an error) because `structopt v0.3.9` was yanked.

```console
$ cargo install --path . --locked
  Installing cargo-udeps v0.1.8 (/home/ryo/src/github.com/est31/cargo-udeps)
    Updating crates.io index
warning: package `structopt v0.3.9` in Cargo.lock is yanked in registry `crates.io`, consider running without --locked
    Finished release [optimized] target(s) in 0.57s
   Replacing /home/ryo/.cargo/bin/cargo-udeps
    Replaced package `cargo-udeps v0.1.8 (/home/ryo/src/github.com/est31/cargo-udeps)` with `cargo-udeps v0.1.8 (/home/ryo/src/github.com/est31/cargo-udeps)` (executable `cargo-udeps`)
```

qryxip commented:
`failure v0.1.7` has just been published.

est31 commented on .github/workflows/ci.yml:
```diff
@@ -55,19 +55,36 @@ jobs:
+          command: install
+          args: --path . --debug --verbose --locked
+
+      # https://github.com/rust-lang-nursery/failure/issues/342
```
Can you remove this comment and activate the section below?

qryxip replied:
Ok. I activated the section.

est31 commented:
Hmm I can't merge, travis seems to be broken.

qryxip commented:
Passed with this...

```diff
+curl-sys = "=0.4.27"
```

est31 commented:
> +curl-sys = "=0.4.27"

We should debug the issue a little and then file a bug at https://github.com/alexcrichton/curl-rust .
